test/test-viz.cpp

```diff
@@ -0,0 +1,350 @@
+        }
+    }
+}
+
+void setConnections(Vertex& element, const std::vector<std::vector<Vertex>> elements, std::vector<Line>& connections) {
+    for(auto input : element.ins) {
+        for(auto layered_elements : elements) {
+            for(auto compared_element : layered_elements) {
+                if(input.name == compared_element.name) {
+                    Line connect;
+                    connect.x1 = compared_element.position.x + compared_element.position.w;
+                    connect.y1 = compared_element.position.y + compared_element.position.h/2;
+                    connect.x2 = element.position.x;
+                    connect.y2 = element.position.y + element.position.h/2;
+                    connections.emplace_back(connect);
+                }
+            }
+        }
+    }
+}
+
+int main(int argc, char* argv[]) {
+
+    if(argc != 2) return -1;
+
+    std::string filename(argv[1]);
+
+    logic_scheme logicScheme;
+
+    bench_parser parser(logicScheme);
+    auto result = lorina::read_bench(filename, parser);
+    if(result != lorina::return_code::success) return -2;
+
+    //Clearing the queue
+    logicScheme.request_handling();
+    logicScheme.output();
+
+    SDL_Window* window = nullptr;
+    SDL_Renderer* renderer = nullptr;
+    std::vector<Line> connections;
+    if(SDL_Init(SDL_INIT_VIDEO) < 0) return -3;
+    window = SDL_CreateWindow("test-viz", 0, 0, SCREEN_WIDTH, SCREEN_HEIGHT, SDL_WINDOW_SHOWN);
+    renderer = SDL_CreateRenderer(window, -1, SDL_RENDERER_ACCELERATED);
+
+    //Finding total number of layers in element with max layer
+    int layersNum = 0;
+    for(auto element : logicScheme._scheme) {
+        if(element.layer > layersNum) layersNum = element.layer;
+    }
+    layersNum++;
+
+    //Sorting elements in order by layers
+    std::vector<std::vector<Vertex>> elements_by_layers(layersNum);
+    for(Vertex& element : logicScheme._scheme) {
+        elements_by_layers[element.layer].emplace_back(element);
+    }
+
+    //Finding the number of elements in layer with the most elements
+    int maxElems = 0;
+    for(auto elements : elements_by_layers) {
+        if(elements.size() > maxElems) {
+            maxElems = elements.size();
+        }
+    }
+
+    const float rectHeight = SCREEN_HEIGHT/(VER_SPACING*maxElems);
+    const float rectWidth = SCREEN_WIDTH/(HOR_SPACING*layersNum);
+
+    std::cout << rectWidth << ' ' << rectHeight;
+
+    for(std::vector<Vertex>& elements : elements_by_layers) {
+        setPosition(elements, maxElems, rectHeight, rectWidth);
+    }
+
+    std::vector<Line> connection_lines;
+    for(std::vector<Vertex>& elements : elements_by_layers) {
+        for(Vertex& element : elements) {
+            setConnections(element, elements_by_layers, connection_lines);
+        }
+    }
+
+    SDL_SetRenderDrawColor(renderer, 0,0,0, SDL_ALPHA_OPAQUE);
+    SDL_RenderClear(renderer);
+    SDL_SetRenderDrawColor(renderer, 255,255,255, SDL_ALPHA_OPAQUE);
+
```

Comment:
Do not mix SDL setup and placement logic: create corresponding procedures and keep them separate.